### PR TITLE
feat(layout): add Position struct

### PR DESCRIPTION
**Pull request by joshka**
This stores the x and y coordinates (columns and rows)

- add conversions from Rect
- add conversion with Size to Rect
- add Rect::as_position

### CODE REVIEW

File: src/layout/position.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
+#![warn(missing_docs)]
+use crate::layout::Rect;
+
+/// Position in the terminal
+///
+/// The position is relative to the top left corner of the terminal window, with the top left corner
+/// being (0, 0). The x axis is horizontal increasing to the right, and the y axis is vertical
+/// increasing downwards.
+pub struct Position {
```
Comment (Valentin271) [inline src/layout/position.rs:9]: Could you add the derive macros here

**Review by Valentin271 — CHANGES_REQUESTED** [review]

**Comment by Valentin271** [comment]
Could you also add a test for `From<(Position, size)> for Rect`? (don't forget to fix formatting)

**Comment by Valentin271** [comment]
Check out https://github.com/ratatui-org/ratatui/pull/789#issuecomment-1887535265.

Mainly this allows easy manipulation and especially storage of positions.

**Comment by joshka** [comment]
One place I'd like this is instead of a (u16,u16) tuple in a state object

**Comment by joshka** [comment]
realized I pushed, but didn't see that it failed due to the a missing import in a doc test. Pushed now

**Comment by Valentin271** [comment]
It seems like the doc-tests are still failing due to missing import @joshka.

**Comment by joshka** [comment]
> It seems like the doc-tests are still failing due to missing import @joshka.

Missed the commit. Fixed now.

**Review by Valentin271 — APPROVED** [review]

**Pull request merged**